usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -489,6 +519,467 @@ static CK_RV cleanse_attribute(TEMPLATE *template,
     return CKR_OK;
 }
 
+/*******************************************************************************
+ *
+ *                    Begin EP11 protected key option
+ */
+
+typedef struct {
+    CK_BBOOL wrap_was_successful;
+    CK_VOID_PTR secure_key;
+    CK_ULONG secure_key_len;
+    CK_BYTE *pkey_buf;
+    size_t *pkey_buflen_p;
+} pkey_wrap_handler_data_t;
+
+/* A wrapped key can have max 132 bytes for an EC-p521 key */
+#define MAX_WRAPPED_KEY_SIZE    (2 * (521 / 8 + 1))
+typedef struct {
+    uint16_t version;
```

Comment:
Are there any defines for the version?

Reply:
No. There is currently only one version 0x0001 supported/returned by zFirmware. But I see what you mean ... maybe it's better to also check this version field in routine ep11tok_pkey_skey2pkey() before extracting the protected key blob. I guess that the length of the whole wrapped_key_t struct would also change when they ever introduce a new version. So the check must be done together with the length check ....

Reply:
_ok, I added the check and introduced a new constant `EP11_WRAPPED_KEY_VERSION_1       0x0001`